cluster-autoscaler/cloudprovider/POLICY.md

```diff
@@ -0,0 +1,127 @@
+document describes the role and responsibilities of core maintainers and
+integration owners. A lot of what is described below has been unofficial
+practice for multiple years now, but this policy also introduces some new
+requirements for cloudprovider maintenance.
+
+## Responsbilities
+
+Cloudprovider owners are responsible for:
+
+  * Maintaining their integrations.
+  * Testing their integrations. Currently any new CA release is tested e2e on
+    GCE, testing on other platforms is the responsibility of cloudprovider
+    maintainers (note: there is an effort to make automated e2e tests possible
+    to run on other providers, so this may improve in the future).
+  * Addressing any issues raised in autoscaler github repository related to a
+    given provider.
+  * Reviewing any pull requests to their cloudprovider.
+    * Pull requests that only change cloudprovider code do not require any
+      review or approval from core maintainers.
+    * Pull requests that change cloudprovider and core code require approval
+      from both the cloudprovider owner and core maintainer.
+
+The core maintainers will generally not interfere with cloudprovider
+development, but they may take the following actions without seeking approval
+from cloudprovider owners:
+
+  * Make trivial changes to cloudproviders when needed to implement changes in
+    CA core (ex. updating function signatures when a go interface
+    changes).
+  * Revert any pull requests that break tests, prevent CA from compiling, etc.
+
+## Adding new cloud provider integration
+
+In order to add new integration you need to open a pull request implementing
+the interfaces defined in cloud\_provider.go. This policy requires that any new
+cloudprovider follows the following rules:
+
+  * Cloudprovider needs to have an OWNERS file that lists its maintainers.
+    Kubernetes policy requires that code OWNERS are members of the Kubernetes
+    organization.
+    * This can create a chicken and egg problem, where adding a cloudprovider
+    requires being a member of Kubernetes org and becoming a member of the
+    organization requires a history of code contributions. For this reason it
+    is allowed for the OWNERS file to temporarily contain
+    commented out github handles. There is an expectation that at least some of
+    the owners will ultimately join Kubernetes organization (by following the
+    [process](https://github.com/kubernetes/community/blob/master/community-membership.md))
+    so that they can approve PRs to their cloudprovider.
+  * Cloudprovider shouldn't introduce new dependencies (such as clients/SDKs)
+    to top-level go.mod vendor, unless those dependencies are already imported
+    by kubernetes/kubernetes repository and the same version of the library is
+    used by CA and Kubernetes. This requirement is mainly driven by
+    the problems with version conflicts in transitive dependencies we've
+    experienced in the past.
```

Comment:
is it worth mentioning that cloudproviders are welcome to carry dependencies in their directory if they need?

Reply:
done